### PR TITLE
Feature/pipeline waves 274

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:* 274

*Description of changes:* Waves support with pipelines to deploy a group of clusters in parallel.

Review is needed only for https://github.com/aws-quickstart/ssp-amazon-eks/pull/286/commits/6c624bb15f2c6f3370d5c99b415eafbff6a2e76e

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shapirov103** [comment]
I ran preliminary tests. Functional test through codepipeline requires published artifact.

**Comment by shapirov103** [comment]
will merge the csi secrets first.

**Comment by shapirov103** [comment]
Please review just this commit: https://github.com/aws-quickstart/ssp-amazon-eks/pull/286/commits/6c624bb15f2c6f3370d5c99b415eafbff6a2e76e

**Review by parkand1 — CHANGES_REQUESTED** [review]
Left comments on the language but asides from that looks good to me

File: lib/pipelines/code-pipeline.ts [hunk 1]
```diff
@@ -54,7 +54,12 @@ export type PipelineProps = {
     /**
      * Pipeline stages and options.
      */
-    stages: StackStage[];
+    stages: WaveStage[];
+
+    /**
+     * Waves for the pipe;ine. Stages inside the wave are executed in parallel.
```
Comment (askulkarni2) [inline lib/pipelines/code-pipeline.ts:60]: Minor: typo pipe;ine

**Review by askulkarni2 — COMMENTED** [review]
It seems like code from Secrets Store CSI refactor is mixed in here (https://github.com/aws-quickstart/ssp-amazon-eks/commit/ce3bd6e0b308ef86cf5e893fab5f6917bf8755a2) is  mixed in here. Am I missing something?

**Comment by shapirov103** [comment]
> It seems like code from Secrets Store CSI refactor is mixed in here ([ce3bd6e](https://github.com/aws-quickstart/ssp-amazon-eks/commit/ce3bd6e0b308ef86cf5e893fab5f6917bf8755a2)) is mixed in here. Am I missing something?

Yes, it is. I have branched off the argocd csi secrets branch, hoping these commits will not show up as they are part of the main already. See the PR description, I left a comment on it.

**Comment by shapirov103** [comment]
e2e succeeded. however, unsure why the PR is not showing it.

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Review by youngjeong46 — APPROVED** [review]
LGTM!

**Review by parkand1 — APPROVED** [review]
LGTM

**Pull request merged**